### PR TITLE
Develop Google add-on

### PR DESCRIPTION
**Pull request by ipitah**
Develop Google add-on

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by ipitah** [comment]
@ninjaginja Find the article here.

**Comment by ninjaginja** [comment]
great, next time you can simply link the issue to the PR using the right menu bar (where it says 'Linked Issues')

**Comment by ipitah** [comment]
@ninjaginja That is well noted

**Comment by kayere** [comment]
Hello @ipitah. Awesome first article. I'll be reviewing your article. In the meantime, please add your author details. 
Notify me when you are done.

**Comment by ipitah** [comment]
@kayere Thank you, my first article was approved however, it has taken some time to be published which explains why you cannot find my author details, will inform you once effected. You can check it here https://github.com/section-io/engineering-education/pull/849

**Comment by kayere** [comment]
Ooow .. Thanks for the info. I am sure your article will be published soon enough. 
Let me start the review.

**Comment by ipitah** [comment]
@kayere The first article was published

File: articles/develop-google-add-on/index.md [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+```javascript
+// Google Drive
+ DriveApp;
+
+//  Doc
+ DocumentApp;
+```
+
+##### 11. Publishing
+This is the last step that allows developers to make available the add-on to the public or targeted group. The add-on should be properly tested for the specified type i.e. Workspace or Editors add-on. The project version is created and the platform while verifying collaborators.
```
Comment (kayere) [inline articles/develop-google-add-on/index.md:214]:
```suggestion
This is the last step. It allows developers to make the add-on available to the public or targeted group. The add-on should be properly tested for the specified type i.e. Workspace or Editors add-on. The project version is created and the platform while verifying collaborators.
```

**Review by kayere — COMMENTED** [review]
Hello @ipitah, I have done a preliminary revision to your article and I have the following suggestions. Please make the changes.

File: articles/develop-google-add-on/index.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
+
+### Things to learn to Develop an Addon in Google Apps Script
+
+#### Introduction
+Google offers its services online through its applications such as Doc, Sheet, Forms, etc. The services collect, hold, and/or store data that need to be shared, processed, or even analyzed. Also, Google has provided a scripting language to allow automation and interaction with its applications. Apps Script is a Javascript cloud language by Google to make automation a reality and make third-party services or information available. The language is useful in building web apps and desktop and android add-ons for Sheets, Docs, and Forms applications. There are two types of add-ons: Google Workspace that works for cross applications and Editor Add-on which works with a single application.  Developers can benefit a lot by being aware of how to create addons. This provides a channel to conveniently offer services to google applications users, using a serverless platform.
```
[the remark this answers is not in the document]
Comment (ipitah) [reply articles/develop-google-add-on/index.md:5]: Captured

File: articles/develop-google-add-on/index.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+The Apps Script displays a programming interface as shown below.
+
+![Apps Script Editor](/engineering-education/google-sheet-add-on/apps-script-editor.png)
+
+We now head straight to the features and development.
+
+##### 1. Menu creation
+The aim is to add on the google sheet menu our functions for users to click and execute a script. The script remains invisible for users hence, the need to have menu buttons to enable interaction and execution. To create a menu onOpen() function is used with the reference to the spreadsheet's user interface being made in the body. A function createMenu('Menu') is called and provided with a suitable name for the add-on top Menu. To add items on the menu, the addItem('Item') function is used with a suitable name provided. Items in a menu can be separated by calling a function addSeparator() that does not take any parameter.  To add a submenu, the addSubMenu() function is called that reference also the spreadsheet's user interface and call the createMenu() function supplying it with the submenu name. The function addToUi() is used to tell the spreadsheet to add the menu in the user interface.
```
[the remark this answers is not in the document]
Comment (ipitah) [reply articles/develop-google-add-on/index.md:19]: Noted

File: articles/develop-google-add-on/index.md [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    .setTitle('Returned Data');
+SpreadsheetApp.getUi().showSidebar(htmlOutput);
+
+}
+```
+
+![Helper function](/engineering-education/google-sheet-add-on/helperFunction.png)
+
+##### 5. Triggers
+Add-ons are capable of using three types of triggers: simple triggers that run on performing a certain function like the onOpen() that is called when the spreadsheet is opened; installable triggers that are called after being installed like the form-submit; time-driven triggers that are time-dependent. In this section, we illustrate the time-based triggers which are defined mostly by the user. They let the script execute at a particular time at most once per hour. You specify the time you want the script to execute but the script may randomly choose the time within that hour which is appropriate to execute i.e. if you choose to execute at 8 am, the script may choose between 8 am and 9 am.
```
[the remark this answers is not in the document]
Comment (ipitah) [reply articles/develop-google-add-on/index.md:105]: Noted

**Comment by ipitah** [comment]
@kayere 
Kindly find the revised article.

**Review by kayere — APPROVED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Pull request merged**